tests/e2e/main_test.go

```diff
@@ -94,13 +95,16 @@ func (s *BabylonSDKTestSuite) Test1ContractDeployment() {
 	s.NoError(err)
 	s.Equal(adminResp["admin"], s.ConsumerCli.GetSender().String())
 
-	// update the contract address in parameters (typically this has to be done via gov props)
-	ctx := s.ConsumerChain.GetContext()
-	params := s.ConsumerApp.BabylonKeeper.GetParams(ctx)
-	params.BabylonContractAddress = s.ConsumerContract.Babylon.String()
-	params.BtcStakingContractAddress = s.ConsumerContract.BTCStaking.String()
-	err = s.ConsumerApp.BabylonKeeper.SetParams(ctx, params)
-	s.NoError(err)
+	// update the contract address in parameters
+	msgUpdateParams := &bbntypes.MsgUpdateParams{
```

Comment:
i see, interesting so contract address are inherently module params which cosmwasm enabled chains would use later to trigger the contract (doSudoCall)?

Reply:
btw iiuc these are just example implementations correct? In practice the consumer chains would need to implement this in their code or we would do that on their behalf and raise a pr?

Reply:
> i see, interesting so contract address are inherently module params which cosmwasm enabled chains would use later to trigger the contract (doSudoCall)?

exactly.

> btw iiuc these are just example implementations correct? In practice the consumer chains would need to implement this in their code or we would do that on their behalf and raise a pr?

This will serve both as a reference implementation and as a production ready implementation for Cosmos SDK chains. If the consumer chain is based on Cosmos SDK and is willing to use this module directly then they can go for it. Otherwise they can build their own middleware between the node and the contracts. A PR is needed for sure.